report/5.Solution/DockerBuild.tex

```diff
@@ -0,0 +1,107 @@
+\section{Build process}\label{sec:buildprocess}
+In previous semesters, the build process has been straightforward, as it has simply been a matter of using the tool included with the IDE of choice for the chosen language.
+This semester the build process is quite a bit more complicated as the project targets the NXT.
+This complicates the build process for a number of reasons:
+nxtOSEK is the operating system for RTS on the NXT, and even though it is quite old, the latest release, as of the writing of this report, was released in January of 2013\cite{osekrelease}.
+
+During the groups testing of the nxtOSEK build, it was found that it was largely incompatible with modern operating systems, both Windows, OS X and a number of modern Linux distributions.
+According to the latest documentation on the nxtOSEK webpage, the latest supported version of Linux is Ubuntu 11.10 which was released in 2011.
+This complicated things a bit, as none of the group members were interested in installing such an old operating system on their personal computer, and instead Docker was used to avoid this.
+
+\subsection{Docker}\label{subsec:docker}
+Docker is a way to deploy a consistent environment for applications to run in, also known as containerization\cite{dockerdocstart}.
+A Docker deployment consists of two different concepts, the container and an image.
+\begin{description}
+    \item [Images] are executables that includes all the required information to instantiate containers.
+    \item[Containers] are the runtime instances of an image, the running process of an image.
+\end{description}
+
+A Docker container is in broad strokes the same as a virtual machine, although a Docker container only uses the required resources, while a virtual machine is allowed access to more resources than is typically needed.
+Docker runs natively on Linux, and takes no more memory than other executables.
+
+A Docker deployment consists of two different concepts, the container and an image.
+\begin{description}
+    \item[Images] are executables that includes all the required information to instantiate containers.
+    \item[Containers] are the runtime instances of an image, the running process of an image.
+\end{description}
+The concept of containerization as a way to ensure consistency in deployment of software, was perfect for the purposes of the group.
+Docker is supported on all 3 major platform, and while both Docker for Windows and OSX run in a virtual machine, this is a non-issue.
+
+A potential issue was nxtOSEK not being officially supported past Ubuntu 11.10, and Docker officially supporting Ubuntu 18.04, 16.04 and 14.04 /cite{dockerubuntu}.
+However, a quick test proved that nxtOSEK still worked fine on Ubuntu 14.04.
```

Comment:
However, using some old dependencies and with a bit of modification it proved to be fairly straightforward to run on Ubuntu 14.04 --- or something like that, det krævede lidt tweaking og downgrading af dependencies. --- "Quick test"... fuck off :stuck_out_tongue_closed_eyes: